### DISCUSSION TITLE
Does this still work on GOG.

### DISCUSSION THREAD

**Discussion by Alanray64** [leodium/leodium-ReLive-NoMansSky-Expeditions-Offline#48] [General]
Just tried this today with GOG purchased NMS, had the internet off so started as offline. Before starting I had moved the JSON over ( I used the 1st Expedition). Started the game and checked and the Expedition Quest giver but no, nothing. I had a reference to the current expedition with two weeks on it.
So, am I doing something wrong or is it no longer working?

Thanks

**Comment by cwmonkey** [comment]
If you're still seeing the current expedition while offline, then you're either somehow still online, or wherever you put your .json file is the wrong place.

If you're properly offline, you should see this in the upper left hand corner of the "SELECT SAVE GAME" screen:
![image](https://github.com/user-attachments/assets/ab8d1413-11c4-44fb-accb-01494f418345)

If you're sure you're offline and still see the current expedition after getting rid of the current cache json file, then you'll have to figure out the correct location on your hard drive.

Also keep in mind you need to stay offline for the duration of the time nms is open as it now attempts to update expeditions as soon as it finds a connection.

This issue hasn't really come up, so I'm not sure what to suggest beyond that.

**Reply by Alanray64** [reply]
Hello and thank you for replying. I can confirm I do get that connecting image.

![image](https://github.com/user-attachments/assets/d45a5b7e-f683-42fb-bb55-9539983fa2c8)
The above is where my NMS cache location resides. 
The pathway being C:>Users>AppData>Roaming>HelloGames>NMS>DefaultUser>Cache

The only difference I see withe the pathing and what was shown here was that I have a DefaultUser in there, so not really sure?

Are you suggesting that I may have it installed elsewhere or messed up my original Installation?

Thank you

**Reply by cwmonkey** [reply]
> C:>Users>AppData

There should be a username between `C:\Users\` and `\AppData`. I'd check the `AppData` directories in the other user directories and see if maybe you're putting it in the wrong user's AppData.

**Comment by Alanray64** [comment]
Thanks again for reading. Ok I see the below -

![image](https://github.com/user-attachments/assets/238b03a4-78c0-436e-b25f-b03c2537e2a2)

An when I look within my account name - 

![image](https://github.com/user-attachments/assets/b31ce453-1922-4dfc-9f54-064f072327dd)

I am using Windows 11 if that helps. However there no name holding HelloGames>NMS in there

**Reply by cwmonkey** [reply]
The `AppData` directory is hidden by default. Where does it go if you put in `%AppData%`? You could also try manually typing in the `\AppData` after your username

**Reply by cwmonkey** [reply]
FYI I'm stuck on Windows 10, but everything I am reading says Windows 11 keeps the AppData directory within the username directory, so I'm relatively sure that's your issue. I'm not even sure what `C:\Users\AppData` is.

**Reply by Alanray64** [reply]
Hmm you may be right on this ---

![image](https://github.com/user-attachments/assets/3a134e05-afd6-42ff-a15f-bdaee897bbb4)


And yes you are. I went offline and tried and I saw expedition 1 showing at the terminal.

Many thanks for your assist, I do not know why there is a duplicate, I guess. Still it is working and I can confirm that.

Just to be sure. I am not to go online with NMS till I finish the expedition? I can save when I leave and go back in as per normal, just always being offline?

Thankyou again for your assist and help.

**Reply by cwmonkey** [reply]
Yeah, you can go online to play other saves, but they changed it recently to constantly check to see if it can communicate with their servers, so if you're in an expedition save when you go online it will try to end the expedition.

I actually set up a firewall rule to block just No Man's Sky so I can keep a little window open that lets me turn the rule on and off like so (did this on Windows 10, but should be the same on Windows 11):

1. Open the windows menu
2. Type `WF.msc` and hit enter
3. Click `Outbound Rules`
4. Click `Action` > `New Rule...`
5. `Program` > `Next >`
6. `This program path:` and find `NMS.exe` somewhere like `C:\Program Files (x86)\Steam\steamapps\common`
7. `Block the connection` > `Next >`
8. Probably only `Public` is needed, but doesn't hurt to check them all, `Next >`
9. Name it and click `Finish`
10. You can now open `WF.msc` at any time and go to your `Outbound Rules` to turn your NMS rule on and off easily by checking/unchecking `Enabled` and clicking `OK`/`Apply`:
![image](https://github.com/user-attachments/assets/31d50a1c-07fe-4dd8-bcab-d9fa473540ce)

**Reply by Alanray64** [reply]
Brilliant idea and thanks for this. I won't try it tonight as it is late but I will give it ago tomorrow.

Thanks once more.

**Reply by dmrickey** [reply]
Would setting up a rule to block NMS make it so that steam cloud saves still work? Since in this case I'm not going into offline mode?

**Reply by cwmonkey** [reply]
> Would setting up a rule to block NMS make it so that steam cloud saves still work? Since in this case I'm not going into offline mode?

@dmrickey Yep, Steam works as normal while nms.exe is blocked via firewall

**Reply by NicolasJLA** [reply]
Hello guys, I was using your firewall rules exactly how it is written and everything worked perfectly. I am playing NMS with PC GAME PASS and I was able to complete the first 6 expeditions while playing offline and then this morning my game is back online. I am trying to start expedition 7 but it's not possible as long as i am online so I am a bit confused to what have changed since yesterday ? I tried to delete the rule in wf.msc and create it again. Nothing. I tried also to set up my PC GAME PASS account to online and offline again. Nothing. I don't know how to fix that. To note there was a windows 11 update yesterday but I have looked at it and nothing seems to have created something wrong with the firewall rule. PLS HELP

**Reply by cwmonkey** [reply]
@NicolasJLA nothing has changed for a few days. What exactly is the issue you're having? What are you seeing/not seeing and what are you expecting?

**Reply by NicolasJLA** [reply]
thx for the quick reply. My issue is the fact htat my game is back to online mode. I am seeing players base for example. Because of that I think the game update the Json file and I am not able to start the expedition 7 when i go to the anomaly or even if i tried to start it from the main menu (new game).
I think the problem is not from my rule because i did not touch it and yesterday the game was offline. I finished expedition 6 the day before and went to my normal save, did a bit of things in my bases, nothing crazy, finished a quest for the egg living ship etc... and this morning as i started my game, it was back to online.

**Reply by cwmonkey** [reply]
Yeah that's definitely not possible assuming the firewall rule is working, so it must not be for some reason. Not really sure what to check, maybe try rebooting and turning the rule on and off, etc.

**Reply by NicolasJLA** [reply]
I tried already but i will try again
This is my path in the rule
C:\XboxGames\No Man's Sky\Content\Binaries\NMS.exe
I think it correct since it was working fine before he morning. And everything is set to be blocked (Domain/private/public)

**Reply by cwmonkey** [reply]
Maybe also try deleting the rule and remaking it and/or try blocking other programs to see if it works.

**Reply by NicolasJLA** [reply]
Nothing. I started the game and as expected it has change the json file (i check it after) it is not the one i have downloaded and moved into my cache file. So the problem is really the fact that my rule is not working. I am clueless about it. I don't understand.

**Reply by cwmonkey** [reply]
@NicolasJLA did you ever get this working? In case you missed this message:

> Maybe also try deleting the rule and remaking it and/or try blocking other programs to see if it works.

**Reply by NicolasJLA** [reply]
No it did not. I ended up cutting completely the wifi connection on my computer but it is not a good solution even if it's working. I even called a friend working in IT like me but she is much more technology savy

**Reply by cwmonkey** [reply]
@NicolasJLA Did they have you try deleting the firewall rule and remaking it from scratch?

I would also try making a firewall rule to block something else like Google Chrome to make sure your firewall rules are actually working.

I'd also verify the correct exe path is being used by running No Man's Sky, opening the task manager, clicking the > next to the running app, right clicking the process and clicking "properties".

**Reply by NicolasJLA** [reply]
Yes I did that and it seems the problem is as you say from the firewall rules. Also everything was working as expected before the windows 11 update. The rule and the path was 100% correct. I was able to play offline and able to start previous expeditions. Again everything was good. I did expeditions 1 to 6 without any issue. Thx btw for the amazing tools 👍. I am running offline with wifi off at the moment so later today I will try again your suggestions but I don't want to fuck up my save so I will not play again with rules until I am sure it's working and my expedition 7 completed

**Comment by cwmonkey** [comment]
@NicolasJLA Aight. Could be that some other program took over for Windows's firewall for some reason. If so you could look into ho to disable it or set up a firewall rule in the other app.

The last thing I would suggest/try is resetting the windows firewall settings to default, either through the interface or by using this command: `netsh advfirewall reset` - you can of course backup your settings just in case.

From there your apps will start asking permission to access the internet again, and you can either disallow nms or you can allow it then try setting up a rule again if the reset actually fixes the problem rules being ignored.

**Reply by NicolasJLA** [reply]
Hi, good news it's working again. It was Norton the problem. It was taking over windows firewall so I deleted it (it's useless anyway) and then the firewall rules was active again and everything back to what it was. All good then. Thx the help
